dali/operators/util/randomizer.cuh

```diff
@@ -64,6 +64,70 @@ struct curand_normal_dist<double> {
   }
 };
 
+
+template <typename T>
+struct curand_uniform_dist {};
+
```

Comment:
How about creating
```c++
template<typename T>
struct curand_uniform_dist {
static_assert( /* check if T is double or float */ );
DALI_HOST_DEV curand_uniform_dist(T start, T end)
/* etc... */
};
```
And specializing only `curand_uniform`/`curand_uniform_double`? This way we have this extra static verification of type with meaningful error msg (and less code duplication). Cause after
```c++
curand_uniform_dist<int> d;
```
We're getting something peculiar I guess

Reply:
Fixed. I changed it to a forward declaration, so it won't compile for other types.